staging/src/k8s.io/apimachinery/pkg/apis/meta/internalversion/types.go

```diff
@@ -48,6 +47,9 @@ type ListOptions struct {
 	ResourceVersion string
 	// Timeout for the list/watch call.
 	TimeoutSeconds *int64
+
+	Limit    int64
```

Comment:
There's client side code (that I would change by default in kube) that can't safely be enabled until we go beta.  So for the 1.8 release cycle the client side paging for informers can be on for testing, but it must be disabled before we ship 1.8 (so that if in 1.9 we realize the alpha api has to change, we don't have a big set of clients using it by default and rolling updates aren't compromised).